oblast/src/lib.rs

```diff
@@ -0,0 +1,223 @@
+//! High-level wrapper for BLS12-381 arithmetic using `blst`.
+
+#[cfg(test)]
+mod tests;
+mod constants::MODULUS_BIT_SIZE;
+
+use blst::{blst_fp12, blst_fr, blst_scalar};
+use paste::paste;
+
+/// Field sub-group element.
+#[derive(Debug, Default, Copy, Clone, PartialEq, Eq)]
+pub struct Fr {
+    element: blst_fr,
+}
+
+impl Fr {
+    pub fn from_u64(value: u64) -> Self {
+        let mut point = Self::default();
+        let input = vec![value, 0, 0, 0];
+        unsafe {
+            blst::blst_fr_from_uint64(&mut point.element, input.as_ptr());
+        }
+        point
+    }
+
+    pub fn from_raw(element: blst_fr) -> Self {
```

Comment:
ah this is nice -- gives the caller the ability to provide something greater than a `u64` for the `Fr` initialization...

we will also probably want a `{to,from}_bytes` method but we can add this down the line